cpp/oneapi/dal/io/csv/detail/read_graph_kernel_impl.hpp

```diff
@@ -39,28 +39,38 @@ inline void read_edge_list(const std::string &name, edge_list<std::int32_t> &eli
     if (!file.is_open()) {
         throw invalid_argument(dal::detail::error_messages::file_not_found());
     }
-
-    elist.reserve(1024);
-    char source_vertex[32], destination_vertex[32];
-    while (file >> source_vertex >> destination_vertex) {
-        auto edge = std::make_pair(daal_string_to_int(&source_vertex[0], 0),
-                                   daal_string_to_int(&destination_vertex[0], 0));
-
-        if (edge.first == 0 && source_vertex[0] != '0') {
-            throw invalid_argument(
-                dal::detail::error_messages::non_numeric_character_in_edge_list());
+    elist.reserve(1024); 
+    std::string line;
+    char * source_endptr;
+    char * dest_endptr;
+    const char *edgeline;
+    std::int32_t source_vertex;
+    std::int32_t destination_vertex;
+    while (getline(file, line)) {      
+        if (line.empty()){
+            break;
+        }
+        edgeline = line.c_str();
+        source_vertex = daal_string_to_int(&edgeline[0], &source_endptr);
+        destination_vertex = daal_string_to_int(source_endptr, &dest_endptr);
```

Comment:
Безопасно было бы проверить, что после первого числа не достигли конца строки, потом только читать второе, но это было бы медленнее.
Мне нужно проверить, что должно работать корректно, если на чтение второй вершины отправится '\0.....'. Если какие-то тесты были пройдены, то должно быть так.

Если указатель сдвигается, то, по идее, все символы с начала должны были прочитаны корректно. Нельзя ли как-то упростить тогда проверки (edge.second == 0 && destination_vertex[0] != '0').

Переполнение при этом не проверяется и что будет, если кто-то сотрет перенос строки.
+мне нужно проверить в какой int переводится строка (int64, int32,...)

Reply:
In case of reading '\0.....' for destination vertex daal_string_to_int returns 0 and this case is checked further in [line](https://github.com/oneapi-src/oneDAL/pull/2049/files#diff-fe645a4702d4c0f19bbedcd97dd66ef77c0edb96233f4fb92964d402400277c9R61)

    (edge.second == 0 && destination_vertex[0] != '0') 
has been replaced with 
    (destination_vertex == 0 && source_endptr[1] != '0')  [line](https://github.com/oneapi-src/oneDAL/pull/2049/files#diff-fe645a4702d4c0f19bbedcd97dd66ef77c0edb96233f4fb92964d402400277c9R61

Reply:
Ok. Thanks!